src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs

```diff
@@ -30,6 +30,13 @@ public async Task<List<CalendarEventSummary>> GetCalendarEvents(GetCalendarEvent
         var eventTypes = GenerateEventTypesSql(options.CalendarIds);
         var regions = GenerateRegionsSql(options.RegionIds);
 
+        var keywordSql = string.Empty;
+
+        if (!string.IsNullOrWhiteSpace(options.Keyword))
+        {
+            keywordSql = " INNER JOIN FREETEXTTABLE(CalendarEvent, Title, '" + options.Keyword.Trim() + "') ft ON (CE.ID = ft.[Key]) ";
```

Comment:
This is prone to SQL injection ⚔️  I think it is ok for now, but we need to rethink how we do this. Or we need to put extra checks on the inputs.

Reply:
Lets setup a brainstorm between Dev team and include Al and see what we can do.

Reply:
ok I have changed the code to use freetext when single word, and contains when >1 word (so it needs to find the combination of words in the title as an exact (case-insensitive) match), and also put a filter in the handler to remove all characters except alphanumeric and space, to deal with injection issues.